driver/main.c

```diff
@@ -1357,94 +1357,166 @@ static const unsigned char compat_nas[21] = {
+#endif
+
+#ifdef __NR_connect
+	case SYS_CONNECT:
+		return __NR_connect;
+#endif
+
+#ifdef __NR_getsockname
 	case SYS_GETSOCKNAME:
-		return PPME_SOCKET_GETSOCKNAME_E;
+		return __NR_getsockname;
+#endif
+
+#ifdef __NR_getpeername
 	case SYS_GETPEERNAME:
-		return PPME_SOCKET_GETPEERNAME_E;
-	case SYS_SOCKETPAIR:
-		return PPME_SOCKET_SOCKETPAIR_E;
-	case SYS_SEND:
-		return PPME_SOCKET_SEND_E;
-	case SYS_SENDTO:
-		return PPME_SOCKET_SENDTO_E;
+		return __NR_getpeername;
+#endif
+
+#ifdef __NR_getsockopt
+	case SYS_GETSOCKOPT:
+		return __NR_getsockopt;
+#endif
+
+#ifdef __NR_setsockopt
+	case SYS_SETSOCKOPT:
+		return __NR_setsockopt;
+#endif
+
+#ifdef __NR_recv
 	case SYS_RECV:
-		return PPME_SOCKET_RECV_E;
+		return __NR_recv;
+#endif
+
+#ifdef __NR_recvfrom
 	case SYS_RECVFROM:
-		return PPME_SOCKET_RECVFROM_E;
-	case SYS_SHUTDOWN:
-		return PPME_SOCKET_SHUTDOWN_E;
-	case SYS_SETSOCKOPT:
-		return PPME_SOCKET_SETSOCKOPT_E;
-	case SYS_GETSOCKOPT:
-		return PPME_SOCKET_GETSOCKOPT_E;
-	case SYS_SENDMSG:
-		return PPME_SOCKET_SENDMSG_E;
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 0, 0)
-	case SYS_SENDMMSG:
-		return PPME_SOCKET_SENDMMSG_E;
+		return __NR_recvfrom;
 #endif
+
+#ifdef __NR_recvmsg
 	case SYS_RECVMSG:
-		return PPME_SOCKET_RECVMSG_E;
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+		return __NR_recvmsg;
+#endif
+
+#ifdef __NR_recvmmsg
 	case SYS_RECVMMSG:
-		return PPME_SOCKET_RECVMMSG_E;
+		return __NR_recvmmsg;
+#endif
+
+#ifdef __NR_send
+	case SYS_SEND:
+		return __NR_send;
+#endif
+
+#ifdef __NR_sendto
+	case SYS_SENDTO:
+		return __NR_sendto;
+#endif
+
+#ifdef __NR_sendmsg
+	case SYS_SENDMSG:
+		return __NR_sendmsg;
+#endif
+
+#ifdef __NR_sendmmsg
+	case SYS_SENDMMSG:
+		return __NR_sendmmsg;
+#endif
+
+#ifdef __NR_shutdown
+	case SYS_SHUTDOWN:
+		return __NR_shutdown;
 #endif
-	case SYS_ACCEPT4:
-		return PPME_SOCKET_ACCEPT4_6_E;
 	default:
-		ASSERT(false);
-		return PPME_GENERIC_E;
+		break;
 	}
+
+	return 0;
 }
+
+static int load_socketcall_params(struct event_filler_arguments *filler_args)
+{
+	unsigned long __user args[6] = {};
+	unsigned long __user *scargs;
+	int socketcall_id;
+	ppm_syscall_get_arguments(current, filler_args->regs, args);
+	socketcall_id = args[0];
+	scargs = (unsigned long __user *)args[1];
+
+#ifdef CONFIG_COMPAT
+	if (unlikely(filler_args->compat)) {
+		compat_ulong_t socketcall_args32[6];
+		int j;
+
+		if (unlikely(ppm_copy_from_user(socketcall_args32, compat_ptr(args[1]), compat_nas[socketcall_id])))
+			return -1;
+		for (j = 0; j < 6; ++j)
+			filler_args->args[j] = (unsigned long)socketcall_args32[j];
+	} else {
+#endif
+		if (unlikely(ppm_copy_from_user(filler_args->args, scargs, nas[socketcall_id])))
+			return -1;
+#ifdef CONFIG_COMPAT
+	}
+#endif
+	return 0;
+}
+
 #endif /* _HAS_SOCKETCALL */
 
+static int preload_params(struct event_filler_arguments *filler_args)
+{
+#ifdef _HAS_SOCKETCALL
+	if (filler_args->is_socketcall)
+	{
+		return load_socketcall_params(filler_args);
```

Comment:
If this is a socketcall, load its params; else store inf `filler_args->args` just the syscall params.